eiBoard/src/test/java/dhbw/eiCompany/service/EntryServiceImpTest.java

```diff
@@ -0,0 +1,174 @@
+        testEntry2.setDate(LocalDate.now().plusDays(1));
+        testEntry2.setDescription("testDescription2");
+        testEntry2.setCategory("testCategory2");
+        testEntry2.setTypId("testTypId2");
+
+        testEntry3.setName("testName3");
+        testEntry3.setPerson(testPerson1);
+        testEntry3.setColor(Color.RED);
+        testEntry3.setDate(LocalDate.now().plusDays(1));
+        testEntry3.setDescription("testDescription3");
+        testEntry3.setCategory("testCategory3");
+        testEntry3.setTypId("testTypId3");
+    }
+
+    @AfterEach
+    void flush(){
+        entryRepository.flush();
+    }
+
+
+    @Test
+    @Order(1)
+    void save() {
+
+        Entry savedEntry1 = service.save(testEntry1);
+        Entry savedEntry2 = service.save(testEntry2);
+        Entry savedEntry3 = service.save(testEntry3);
+
+
+        // For testEntry1
+        assertEquals(savedEntry1.getEntryId(), testEntry1.getEntryId());
+        assertEquals(savedEntry1.getName(), testEntry1.getName());
+        assertEquals(savedEntry1.getPerson(), testEntry1.getPerson());
+        assertEquals(savedEntry1.getColor(), testEntry1.getColor());
+        assertEquals(savedEntry1.getDate(), testEntry1.getDate());
+        assertEquals(savedEntry1.getDescription(), testEntry1.getDescription());
+        assertEquals(savedEntry1.getCategory(), testEntry1.getCategory());
+        assertEquals(savedEntry1.getTypId(), testEntry1.getTypId());
+
+        // For testEntry2
+        assertEquals(savedEntry2.getEntryId(), testEntry2.getEntryId());
+        assertEquals(savedEntry2.getName(), testEntry2.getName());
+        assertEquals(savedEntry2.getPerson(), testEntry2.getPerson());
+        assertEquals(savedEntry2.getColor(), testEntry2.getColor());
+        assertEquals(savedEntry2.getDate(), testEntry2.getDate());
+        assertEquals(savedEntry2.getDescription(), testEntry2.getDescription());
+        assertEquals(savedEntry2.getCategory(), testEntry2.getCategory());
+        assertEquals(savedEntry2.getTypId(), testEntry2.getTypId());
+
+
+    }
+
+    @Test
+    @Order(2)
+    void delete() {
+
+        //Hier wird Liste erstellt nachdem alle Objekte hinzugefügt wurden
+        List<Entry> listBeforeDelete = service.getAllEntries();
+        assertEquals(5, listBeforeDelete.size());
+
+        service.delete(7L);
+
+        /*
+          Anschließend wird Liste nachdem Löschen eines Eintrags ausgegeben und überprüft
+          ob die Liste sich verkleinert hat und ob eine Exception kommt wenn man nach dem
+          gelöschten Objekt sucht
+         */
+
+        List<Entry> listAfterDelete = service.getAllEntries();
+
+        assertEquals(4, listAfterDelete.size());
+
+        assertThrows(NoSuchElementException.class, ()->{
+            service.findById(7L);
+        });
+    }
+
+    @Test
+    @Order(3)
+    void findById() {
+
+        Entry result = service.findById(8L);
+        Optional<Entry> searchedEntry = entryRepository.findById(8L);
+
```

Comment:
Angenommen man macht das richtig mockt man glaube ich alles. Also wenn service.findById aufgerufen wird, das ein gemocktes "Fake Objekt" zurück gegeben wird.
Also nur im allgemeinen, damit man in den Tests nicht mit den wirklichen Daten der Datenbank spielt. In der Praxis wäre das ja Problematisch wenn beim Kunden aufeinmal echte Daten bearbeitet werden.

Ich denke hier könntest du auch dein Entry1 oder so in der Datenbank speichern, das dann mit service.findById dir holen und die beiden Entrys dann vergleichen. (alternativ auch einfach ein Entry objekt anlegen welches die gleichen Inhalte hat wie das mit der ID 8, also auch id 8 etc. und dann schauen ob findById(8) das gleiche rausgibt, allerdings sollten die beispieldaten mal verschwinden würde der Test fehlschlagen..)